dp/cloud/python/magma/radio_controller/services/active_mode_controller/service.py

```diff
@@ -152,11 +148,43 @@ def _list_cbsds(session: Session) -> State:
             contains_eager(DBCbsd.grants).
             joinedload(DBGrant.state),
         ).
-        filter(*query_filter).
+        filter(_build_filter()).
         populate_existing()
     )
 
 
+def _build_filter():
+    multi_step = [
+        DBCbsd.fcc_id, DBCbsd.user_id, DBCbsd.number_of_ports,
+        DBCbsd.min_power, DBCbsd.max_power, DBCbsd.antenna_gain,
+    ]
+    single_step = [
```

Comment:
maybe split it between type A and B single step, since some fields required for B are optional for A

Reply:
as discussed, let's leave it like this for now